plugin/utiltools.cpp

```diff
@@ -0,0 +1,107 @@
+#include "utiltools.h"
+
+#include <QDir>
+#include <QDebug>
+#include <qqmlinfo.h>
+
+static QString SystemTool("sailfish_tools_system_action");
+
+UtilTools::UtilTools(QObject *parent)
+    : QObject(parent)
+{
+}
+
+UtilTools::~UtilTools()
+{
+}
+
+void UtilTools::removeBackups(QJSValue successCallback, QJSValue errorCallback)
+{
+    QDir vaultDir(QDir::homePath() + "/.vault");
+
+    if (vaultDir.removeRecursively()) {
+        if (successCallback.isCallable()) {
+            QJSValue result = successCallback.call();
+            if (result.isError()) {
+                qmlInfo(this) << "Error executing callback";
+            }
+        }
+    } else {
+        if (errorCallback.isCallable()) {
+            QJSValue result = errorCallback.call();
+            if (result.isError()) {
+                qmlInfo(this) << "Error executing error callback";
+            }
+        }
+    }
+}
+
+void UtilTools::cleanRpmDb(QJSValue successCallback, QJSValue errorCallback)
+{
+    execute(SystemTool, QStringList("repair_rpm_db"), successCallback, errorCallback);
+}
+
+void UtilTools::restartKeyboard(QJSValue successCallback, QJSValue errorCallback)
+{
+    QStringList arguments;
+    arguments << "--user" << "restart" << "maliit-server.service";
```

Comment:
Works without, but was already like that and as code I guess this doesn't need the abbreviated format as much as command line usage.